packages/perspective/test/js/sort.js

```diff
@@ -14,8 +14,338 @@ const data = {
+                            x: [1, 2, 3, 4, 4, 3, 2, 1],
+                            y: ["a", "a", "a", "a", "b", "b", "b", "b"]
+                        });
+                        var view = table.view({
+                            columns: ["x", "w"],
+                            row_pivots: ["y"],
+                            aggregates: {
+                                w: "sum"
+                            },
+                            sort: [
+                                ["x", "desc"],
+                                ["w", "desc"]
+                            ]
+                        });
+
+                        const json = await view.to_columns();
+                        expect(json).toEqual({
+                            __ROW_PATH__: [[], ["a"], ["b"]],
+                            w: [11.92, 7.76, 4.16],
+                            x: [20, 10, 10]
+                        });
+
+                        view.delete();
+                        table.delete();
+                    });
+
+                    it("unique", async function() {
+                        var table = perspective.table(data3);
+                        var view = table.view({
+                            columns: ["x", "w"],
+                            row_pivots: ["y"],
+                            aggregates: {
+                                w: "unique"
+                            },
+                            sort: [
+                                ["x", "desc"],
+                                ["w", "desc"]
+                            ]
+                        });
+
+                        const json = await view.to_columns();
+                        expect(json).toEqual({
+                            __ROW_PATH__: [[], ["a"], ["b"]],
+                            w: [null, null, null, null],
+                            x: [20, 10, 10]
+                        });
+
+                        view.delete();
+                        table.delete();
+                    });
+
+                    it("avg", async function() {
+                        var table = perspective.table(data3);
+                        var view = table.view({
+                            columns: ["x", "w"],
+                            row_pivots: ["y"],
+                            aggregates: {
+                                w: "avg"
+                            },
+                            sort: [
+                                ["x", "desc"],
+                                ["w", "desc"]
+                            ]
+                        });
+
+                        const json = await view.to_columns();
+                        expect(json).toEqual({
+                            __ROW_PATH__: [[], ["a"], ["b"]],
+                            w: [3, 2, 1],
+                            x: [20, 10, 10]
+                        });
+
+                        // current wrong result:
```

Comment:
@texodus seems to me the avg here is wrong - it should be `avg(3.5, 4.5, null, null)` and `avg(null, null, 1.5, 2.5)`, so 2 and 1 - the total is right but the individual components are wrong here.

Reply:
nvm - it is calculating the avg. of the non-null rows, which is correct.